### PR TITLE
Add proper exit on activityCanceled event from ETOS

### PR DESCRIPTION
**Pull request by fredjn**
### Applicable Issues
Fixes: eiffel-community/etos#85

### Description of the Change
Etos Client not exits properly on activityCanceled event from ETOS, no longer hanging for 24 hours.

### Benefits
We fail faster.

### Possible Drawbacks
None that I can think of.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Fredrik Fristedt <<fredrik.fristedt@axis.com>>

### CODE REVIEW

File: src/etos_client/lib/graphql.py [hunk 1]
```diff
@@ -85,6 +86,28 @@ def request_activity(etos, suite_id):
     return None
 
 
+def request_activity_canceled(etos, activity_id):
+    """Request an activity event from graphql.
+
+    :param etos: ETOS client query etos.
```
Comment (t-persson) [inline src/etos_client/lib/graphql.py:92]: This sentence doesn't make sense to me

File: src/etos_client/lib/graphql.py [hunk 2]
```diff
@@ -85,6 +86,28 @@ def request_activity(etos, suite_id):
...
+    :type etos: :obj:`etos_lib.etos.ETOS`
+    :param suite_id: ID of execution recipe triggering this activity.
+    :type suite_id: str
```
Comment (t-persson) [inline src/etos_client/lib/graphql.py:95]: Input is activity Id not suite id

File: src/etos_client/lib/graphql_queries.py [hunk 3]
```diff
@@ -44,6 +44,20 @@
 }
 """
 
+ACTIVITY_CANCELED = """
+{
+  activityCanceled(search: "{'links.target': '%s'}") {
```
Comment (t-persson) [inline src/etos_client/lib/graphql_queries.py:49]: Might not be necessary, but adding links.type=ACTIVITY_EXECUTION might make this more explicit

**Review by t-persson — CHANGES_REQUESTED** [review]
Looks good! Just some documentation comments.

File: src/etos_client/lib/graphql_queries.py (hunk 3 above)
Comment (fredjn) [reply src/etos_client/lib/graphql_queries.py]: Fixed

File: src/etos_client/lib/graphql.py (hunk 2 above)
Comment (fredjn) [reply src/etos_client/lib/graphql.py]: Fixed

File: src/etos_client/lib/graphql.py (hunk 1 above)
Comment (fredjn) [reply src/etos_client/lib/graphql.py]: Fixed

**Review by andjoe-axis — APPROVED** [review]
+1

**Review by t-persson — APPROVED** [review]

**Pull request merged**